python/ray/tune/tests/test_sample.py

```diff
@@ -1621,6 +1621,59 @@ def testPointsToEvaluateBasicVariantAdvanced(self):
         self.assertTrue(configs[8]["nested"]["random"] == 8.0)
         self.assertTrue(configs[8]["nested"]["dependent"] == -8.0)
 
+    def testPointsToEvaluateBasicVariantFixedParam(self):
+        config = {
+            "a": 1,
+            "b": tune.randint(0, 3),
+        }
+
+        from ray.tune.suggest.basic_variant import BasicVariantGenerator
+
+        # Test whether the initial points of fixed parameters are correctly
+        # verified.
+        searcher = BasicVariantGenerator(points_to_evaluate=[
+            {
+                "a": 1,
+                "b": 2
+            },
+        ])
+        analysis = tune.run(
+            _mock_objective,
+            name="test",
+            config=config,
+            search_alg=searcher,
+            num_samples=2,
+        )
+        configs = [trial.config for trial in analysis.trials]
+
+        self.assertEqual(searcher.total_samples, 2)
+        self.assertEqual(len(configs), searcher.total_samples)
+        self.assertEqual([cfg["a"] for cfg in configs], [1] * 2)
+        self.assertEqual(configs[0]["b"], 2)
+
+        # Test whether correctly throwing warning if the pre-set value of fixed
+        # parameters isn't the same as its initial points
+        searcher = BasicVariantGenerator(points_to_evaluate=[
+            {
+                "a": 2,
+                "b": 2
+            },
+        ])
+
+        with self.assertLogs(
+                "ray.tune.suggest.variant_generator", level="WARN") as cm:
+            analysis = tune.run(
+                _mock_objective,
+                name="test",
+                config=config,
+                search_alg=searcher,
+                num_samples=2,
+            )
+        self.assertEqual(cm.output, [
+            "WARNING:ray.tune.suggest.variant_generator:Pre-set value `2` is "
+            "not equal to the value of parameter `a`: 1"
+        ])
```

Comment:
I think this has to be indented?

Reply:
Sorry, I don't understand what you mean.

Reply:
Sorry, I don't understand what you mean.
Feel free to modify the code if you think there is a problem.

In addition, on my machine, the test passes but it don't print anything during execution (`tune.run` should have printed some info):
```bash
$ pytest test_sample.py -v -k testPointsToEvaluateBasicVariantFixedParam
==================================================================== test session starts =====================================================================
platform linux -- Python 3.8.10, pytest-5.4.3, py-1.10.0, pluggy-0.13.1 -- /home/LAB/chensi/anaconda3/envs/ray2devpy38/bin/python
cachedir: .pytest_cache
rootdir: /home/LAB/chensi/workspace/dev_ray2py38/python
plugins: typeguard-2.11.1, anyio-3.3.0
collected 43 items / 42 deselected / 1 selected                                                                                                              

test_sample.py::SearchSpaceTest::testPointsToEvaluateBasicVariantFixedParam PASSED                                                                     [100%]

====================================================================== warnings summary ======================================================================
ray/tune/tests/test_sample.py::SearchSpaceTest::testPointsToEvaluateBasicVariantFixedParam
ray/tune/tests/test_sample.py::SearchSpaceTest::testPointsToEvaluateBasicVariantFixedParam
ray/tune/tests/test_sample.py::SearchSpaceTest::testPointsToEvaluateBasicVariantFixedParam
ray/tune/tests/test_sample.py::SearchSpaceTest::testPointsToEvaluateBasicVariantFixedParam
  /home/LAB/chensi/anaconda3/envs/ray2devpy38/lib/python3.8/site-packages/ray/tune/syncer.py:234: DeprecationWarning: ray.services.get_node_ip_address has been moved to ray.util.get_node_ip_address. 'ray.services' will be removed after Ray 1.4.
    self.local_ip = services.get_node_ip_address()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======================================================== 1 passed, 42 deselected, 4 warnings in 8.82s ========================================================
```
Maybe I've done something wrong?